renderer/tr_light.cpp

```diff
@@ -1156,31 +1158,32 @@ void R_AddDrawSurf( const srfTriangles_t *tri, const viewEntity_t *space, const
 			drawSurf->sort += eDef->parms.sortOffset;
 	}
 
-	// bumping this offset each time causes surfaces with equal sort orders to still
-	// deterministically draw in the order they are added
-	tr.sortOffset += 0.000001f;
-
-	// if it doesn't fit, resize the list
-	Sys_EnterCriticalSection( CRITICAL_SECTION_TWO );
-	if ( tr.viewDef->numDrawSurfs == tr.viewDef->maxDrawSurfs ) {
-		drawSurf_t	**old = tr.viewDef->drawSurfs;
-		int			count;
+	if (!deferred) {
+		// bumping this offset each time causes surfaces with equal sort orders to still
+		// deterministically draw in the order they are added
+		tr.sortOffset += 0.000001f;
+		// if it doesn't fit, resize the list
+		Sys_EnterCriticalSection( CRITICAL_SECTION_TWO );
```

Comment:
This was added by Duzenko when he worked on the same task. So I think it would be right to replace it with object mutex, like `idSysMutex`.

Reply:
I'll need to check, but there's a good chance that the non-deferred code path never executes in parallel, anyway, so might just be able to remove the synchronization entirely.